### PR TITLE
Menu label translation

### PR DESCRIPTION
**Pull request by pavol-tk**
## Subject
Allow to use parameters/placeholders in (sidebar) menu label.

translations/messages.en.yaml
```yaml
item_label_translation: Translatable label with %placeholder%.
```

Menu builder:
```php
$group->addChild('item', [
    'label' => 'item_label_translation',
    'labelAttributes' => ['%placeholder%' => 'dynamic_value']
]);
```

## Changelog
```markdown
### Fixed
- Allow to use parameters/placeholders in (sidebar) menu label translation.
```

### CODE REVIEW

**Comment by pavol-tk** [comment]
ping @VincentLanglet

**Comment by VincentLanglet** [comment]
Why did you close this @pavol-tk ?

**Comment by pavol-tk** [comment]
sorry, I thought it was merged

**Comment by VincentLanglet** [comment]
Hi @pavol-tk, let's try to merge this PR !
Can you squash your commit ? You'll maybe need to rebase 3.x too.

@sonata-project/contributors, if someone wanna review.

File: docs/reference/advanced_configuration.rst [hunk 1]
```diff
@@ -306,19 +306,19 @@ If you want to use the Tab Menu in a different way, you can replace the Menu Tem
...
-The translation parameters and domain can be customised by using the
-``translation_domain`` and ``translation_parameters`` keys of the extra array
+The label translation parameters and domain can be customised by using the
+``label_translation_parameters`` and ``label_catalogue`` keys of the extra array
 of data associated with the item, respectively::
 
     $menuItem->setExtras([
-        'translation_parameters' => ['myparam' => 'myvalue'],
-        'translation_domain' => 'My domain',
+        'label_translation_parameters' => ['myparam' => 'myvalue'],
```
Comment (core23) [inline docs/reference/advanced_configuration.rst:314]: Why did you change this to `label_catalogue`? Is there a reason why we could not keep `translation_domain` or why the new parameter is not called `label_catalogue_parameters` ?

Comment (VincentLanglet) [reply docs/reference/advanced_configuration.rst:314]: If you look at the code, you can see
```
{%- set translation_domain = item.extra('label_catalogue', 'messages') -%}
```
So `translation_domain` was not working, he just fixed the doc.

IMHO catalogue/label_catalogue/default_catalogue should be deprecated in favor of Symfony-like options. I already created an issue for this https://github.com/sonata-project/SonataAdminBundle/issues/6116

Comment (phansys) [reply docs/reference/advanced_configuration.rst:314]: The changelog should help to understand what is fixed or improved, but it is missing in this PR.
Could you please solve that @pavol-tk? The instructions are available at [CONTRIBUTING.md](https://github.com/sonata-project/SonataAdminBundle/blob/3.x/CONTRIBUTING.md#the-changelog).

**Comment by pavol-tk** [comment]
Changelog added

**Comment by jordisala1991** [comment]
Please can you rebase so all builds passes?

File: src/Resources/views/Menu/sonata_menu.html.twig [hunk 2]
```diff
@@ -47,4 +47,16 @@
     {% endapply %}
 {% endblock %}
 
-{% block label %}{% if is_link is defined and is_link %}{{ icon|default|raw }}{% endif %}{% if options.allow_safe_labels and item.extra('safe_label', false) %}{{ item.label|raw }}{% else %}{{ item.label|trans({}, translation_domain|default('messages')) }}{% endif %}{% endblock %}
+{% block label %}
+    {% apply spaceless %}
+        {%- if is_link is defined and is_link -%}
```
Comment (phansys) [inline src/Resources/views/Menu/sonata_menu.html.twig:52]:
```suggestion
        {%- if is_link|default(false) -%}
```

**Comment by jordisala1991** [comment]
@pavol-tk can you make the suggested change?

**Comment by jordisala1991** [comment]
Thank you @pavol-tk
